File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L121-BestTimeToBuyAndSellStock.java
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0,
            r = 1,
            maxPro = 0,
            profit = 0,
            totLen = prices.length;
        while (r < totLen) {
            if (prices[l] < prices[r]){
                profit = prices[r] - prices[l];
                maxPro = maxPro>profit ? maxPro : profit;
            }
            else {
                l = r;
            }
            r++;
        }
        return maxPro;
    }
}

File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L121-best_time_to_buy_and_sell_stock.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l, r = 0, 1
        maxPro = 0
        while r < len(prices):
            if prices[l] < prices[r]:
                profit = prices[r] - prices[l]
                maxPro = max(maxPro, profit)
            else:
                l = r
            r += 1
        return maxPro
